src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -225,14 +231,51 @@ internal override void GetDisplayData(VSTREEDISPLAYDATA[] pData)
             pData[0].Image = pData[0].SelectedImage = (ushort)StandardGlyphGroup.GlyphLibrary;
         }
 
-        public static IHierarchicalDifferenceCollection ComputeDiffSpans(SourceText oldText, SourceText newText, ITextDifferencingService diffService, CancellationToken cancellationToken)
+        private static IHierarchicalDifferenceCollection ComputeDiffSpans(ITextDifferencingService diffService, TextDocument left, TextDocument right, CancellationToken cancellationToken)
         {
-            var diffResult = diffService.DiffStrings(oldText.ToString(), newText.ToString(), new StringDifferenceOptions()
+            // TODO: it would be nice to have a syntax based differ for presentation here, 
+            //       current way of just using text differ has its own issue, and using syntax differ in compiler that are for incremental parser
+            //       has its own drawbacks.
```

Comment:
@heejaechang We create a FileChange to represent changes to both additional documents and source document. See the constructor for this type, it has parameters of type `TextDocument`, which could be either a source document or an additional document.

Looking at your change it seems you do handle this case as `ContainsBetterDiff` will return false for additional documents..